StackExchange.Redis/StackExchange/Redis/RedisDatabase.cs

```diff
@@ -1579,6 +1579,605 @@ public IEnumerable<SortedSetEntry> SortedSetScan(RedisKey key, RedisValue patter
+                flags,
+                RedisCommand.XDEL,
+                key,
+                messageIds.ToRedisValueArray());
+
+            return ExecuteSync(msg, ResultProcessor.Int64);
+        }
+
+        public Task<long> StreamMessagesDeleteAsync(RedisKey key, string[] messageIds, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = Message.Create(Database,
+                flags,
+                RedisCommand.XDEL,
+                key,
+                messageIds.ToRedisValueArray());
+
+            return ExecuteAsync(msg, ResultProcessor.Int64);
+        }
+
+        public StreamPendingInfo StreamPendingInfoGet(RedisKey key, RedisValue groupName, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = Message.Create(Database, flags, RedisCommand.XPENDING, key, groupName);
+            return ExecuteSync(msg, ResultProcessor.StreamPendingInfo);
+        }
+
+        public Task<StreamPendingInfo> StreamPendingInfoGetAsync(RedisKey key, RedisValue groupName, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = Message.Create(Database, flags, RedisCommand.XPENDING, key, groupName);
+            return ExecuteAsync(msg, ResultProcessor.StreamPendingInfo);
+        }
+
+        public StreamPendingMessageInfo[] StreamPendingMessageInfoGet(RedisKey key, RedisValue groupName, RedisValue minId, RedisValue maxId, int count, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = GetStreamPendingMessagesMessage(key, groupName, minId, maxId, count, RedisValue.Null, flags);
+            return ExecuteSync(msg, ResultProcessor.StreamPendingMessages);
+        }
+
+        public StreamPendingMessageInfo[] StreamPendingMessageInfoGet(RedisKey key, RedisValue groupName, RedisValue minId, RedisValue maxId, int count, RedisValue consumerName, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = GetStreamPendingMessagesMessage(key, groupName, minId, maxId, count, consumerName, flags);
+            return ExecuteSync(msg, ResultProcessor.StreamPendingMessages);
+        }
+
+        public Task<StreamPendingMessageInfo[]> StreamPendingMessageInfoGetAsync(RedisKey key, RedisValue groupName, RedisValue minId, RedisValue maxId, int count, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = GetStreamPendingMessagesMessage(key, groupName, minId, maxId, count, RedisValue.Null, flags);
+            return ExecuteAsync(msg, ResultProcessor.StreamPendingMessages);
+        }
+
+        public Task<StreamPendingMessageInfo[]> StreamPendingMessageInfoGetAsync(RedisKey key, RedisValue groupName, RedisValue minId, RedisValue maxId, int count, RedisValue consumerName, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = GetStreamPendingMessagesMessage(key, groupName, minId, maxId, count, consumerName, flags);
+            return ExecuteAsync(msg, ResultProcessor.StreamPendingMessages);
+        }
```

Comment:
nit: can we keep sync/async pattern on methods here? last 4 aren't paired up

Reply:
I thought there used to be a test that checked this with reflection... If not, I can add

Reply:
I'm not sure I understand this one, there are sync/async methods for the StreamPendingMessageInfoGet operation. Or is the naming too similar to StreamPendingInfoGet?

StreamPendingInfoGet returns a summary of pending information for the stream, whereas StreamPendingMessageInfoGet returns the information for each pending message.

![image](https://user-images.githubusercontent.com/1170315/41947876-2f474832-7988-11e8-80e9-71277c46f044.png)

Reply:
In this case there are 2 issues:
- Ordering (should be method, async equivalent, method, async equivalent) for consistency. 
- I don't think we need 2 sets of methods here - if consumer name is null, we don't return details (just default it to `RedisValue.Null`)
...that gets the same functionality while simplifying the API. We should document this in the usage though.